nautobot_device_lifecycle_mgmt/migrations/0021_migrate_to_core_software_models.py

```diff
@@ -0,0 +1,264 @@
+            documentation_url=dlm_software_version.documentation_url,
+            long_term_support=dlm_software_version.long_term_support,
+            pre_release=dlm_software_version.pre_release,
+            status=status_active,
+            _custom_field_data=dlm_software_version._custom_field_data,
+        )
+        core_software_version.save()
+
+        # Work around created field's auto_now_add behavior
+        CoreSoftwareVersion.objects.filter(id=core_software_version.id).update(created=dlm_software_version.created)
+        core_software_version.refresh_from_db()
+
+        # Migrate "Software on Device" relationships to the Device.software_version foreign key
+        for relationship_association in RelationshipAssociation.objects.filter(
+            relationship__key="device_soft",
+            source_type=core_software_version_ct,
+            source_id=core_software_version.id,
+            destination_type=device_ct,
+        ):
+            device = Device.objects.get(id=relationship_association.destination_id)
+            device.software_version = core_software_version
+            device.save()
+
+        # Migrate "Software on InventoryItem" relationships to the InventoryItem.software_version foreign key
+        for relationship_association in RelationshipAssociation.objects.filter(
+            relationship__key="inventory_item_soft",
+            source_id=core_software_version.id,
+            source_type=core_software_version_ct,
+            destination_type=inventory_item_ct,
+        ):
+            inventory_item = InventoryItem.objects.get(id=relationship_association.destination_id)
+            inventory_item.software_version = core_software_version
+            inventory_item.save()
+
+        # make tag manager available in migration for nautobot.core.models.utils.serialize_object
+        # https://github.com/jazzband/django-taggit/issues/101
+        # https://github.com/jazzband/django-taggit/issues/454
+        core_software_version.tags = TagsManager(
+            through=extras_models.TaggedItem,
+            model=CoreSoftwareVersion,
+            instance=core_software_version,
+            prefetch_cache_name="tags",
+        )
+
+        # Create an object change to document migration
+        ObjectChange.objects.create(
+            action=extras_choices.ObjectChangeActionChoices.ACTION_UPDATE,
+            change_context=extras_choices.ObjectChangeEventContextChoices.CONTEXT_ORM,
+            change_context_detail="Migrated from Nautobot App Device Lifecycle Management",
+            changed_object_id=core_software_version.id,
+            changed_object_type=core_software_version_ct,
+            object_data=serialize_object(core_software_version),
+            object_repr=f"{core_software_version.platform.name} - {core_software_version.version}"[
+                :CHANGELOG_MAX_OBJECT_REPR
+            ],
+            request_id=uuid.uuid4(),
```

Comment:
Consider having a single `request_id` shared across all ObjectChanges from this migration, instead of a different one for each?